tests/ArmTemplates.Tests/Common/ExtensionsTest.cs

```diff
@@ -0,0 +1,42 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using Xunit;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions;
+using FluentAssertions;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Tests.Common
+{
+    public class ExtensionsTest
+    {
+        [Fact]
+        public void GenerateValidResourceNameFromDisplayName_ShouldReturnNull_GivenEmptyString() 
```

Comment:
If you want to use `ParameterNamingHelper` these tests will be helpful for that class!